test/types/cptr/ptr_cast_to_string.chpl

```diff
@@ -0,0 +1,9 @@
+use Regexp;
+
+var mem = c_calloc(uint(8), 4096);
+if mem != c_nil then
+  assert((mem:string).match(compile("0x[0-9a-f]+")).matched);
+var ptr = mem:c_void_ptr;
+assert((mem:string) == (ptr:string));
+assert((c_nil:string) == "(nil)");
```

Comment:
Hmm. For me on Linux (Ubuntu 16.04 and RHEL6), I see `(nil)`, but I see now that, according to the C specification, the `%p` format produces implementation-defined output. Maybe I should remove that test line.